docs/Iniciando com React/Controlled-vs-uncontrolled-components.md

```diff
@@ -0,0 +1,95 @@
+<p align="center">
+  <a href="https://github.com/he4rt/4noobs" target="_blank">
+    <img src="../../assets/global/header-4noobs.svg">
+  </a>
+</p>
+
+# Controlled Components vs Uncontrolled Components
+
+## Controlled Components
+
+Os Controlled Components (componentes controlados) tem o React como responsável por controlar seus valores. O componente se torna controlado quando sobreescrevemos o valor da DOM, por exemplo, ao colocar a propriedade `value` em um componente de `input`. Com isso, nossa fonte de dado confiável é o estado, pois está controlando o componente, logo, se não atualizarmos o estado, a DOM não será atualizada.
+
+A cada vez que o usuário digita algo, o estado é atualizado e por consequência o componente rerenderiza novamente. Por conta disso, validações em tempo real são utilizadas neste tipo de componente. 
+
+```jsx
+const Component = () => {
+  const [name, setName] = useState("")
+
+  return (
+    <input
+      value={name}
+      onChange={event => setName(event.target.value)}
+    />
+  )
+}
+```
+
+Para passar um valor padrão, basta colocar um valor inicial no estado que está controlando o componente.
+
+```diff
+const Component = () => {
+- const [name, setName] = useState("")
++ const [name, setName] = useState("4noobs")
+
+  return (
+    <input
+      value={name}
+      onChange={event => setName(event.target.value)}
+    />
+  )
+}
+```
+
+## Uncontrolled Components
+
+Os Uncontrolled Components (componentes não controlados) tem a DOM como responsável por controlar seus valores, ou seja, quando precisamos de seus valores, precisamos pegar diretamente da DOM. Podemos fazer isso com o uso do hook `useRef`, que utiliza a referência do elemento da DOM para que seja possível acessar seus valores.
```

Comment:
Ta sendo desenvolvido uma seção sobre useRef aqui: https://github.com/he4rt/react4noobs/pull/119

Acha que seria uma boa linkar o useRef pra essa seção? Pra caso do leitor ainda não ter visto sobre, ou queira revisar

Tipo: '... com o uso do hook [useRef](https://github.com/he4rt/react4noobs/pull/119), que utiliza a referência...'

Reply:
Did in the [last ammend](https://github.com/he4rt/react4noobs/pull/120/commits/9dbf96fe79e50bbb8e0db6913bd233c913232dd8)